enginetest/queries/queries.go

```diff
@@ -8190,6 +8190,25 @@ ORDER BY 1;`,
 			{3},
 		},
 	},
+	{
+		Query:    "SELECT JSON_EXTRACT('0.4', '$')",
+		Expected: []sql.Row{{types.MustJSON(`0.4`)}},
+	},
+	{
+		Query:    "SELECT JSON_EXTRACT('0.4', '$') > 0;",
+		Expected: []sql.Row{{true}},
+	},
+	{
+		Query:    "SELECT JSON_EXTRACT('0.4', '$') <= 0;",
+		Expected: []sql.Row{{false}},
+	}, {
+		Query:    "SELECT JSON_EXTRACT('0.4', '$') = 0;",
+		Expected: []sql.Row{{false}},
+	},
+	{
+		Query:    "SELECT JSON_EXTRACT('0.4', '$') = 0.4;",
+		Expected: []sql.Row{{true}},
+	},
```

Comment:
(minor) Good test cases. Might be slightly better to put these into `json_scripts.go`, just to keep the JSON test cases together.